chia/cmds/coin_funcs.py

```diff
@@ -210,6 +210,27 @@ async def async_split(
             print("Wallet not synced. Please wait.")
             return []
 
+        if number_of_coins is None and amount_per_coin is None:
+            print("Must use either -a or -n. For more information run --help.")
+            return []
+
+        if number_of_coins is None:
+            coins = await wallet_client.get_coin_records_by_names([target_coin_id])
+            if len(coins) == 0:
+                print("Could not find target coin.")
+                return []
+            assert coins is not None
+            assert amount_per_coin is not None
+            number_of_coins = uint64(coins[0].coin.amount // amount_per_coin.convert_amount(mojo_per_unit))
+        elif amount_per_coin is None:
+            coins = await wallet_client.get_coin_records_by_names([target_coin_id])
+            if len(coins) == 0:
+                print("Could not find target coin.")
+                return []
+            assert coins is not None
```

Comment:
same here, about checking for `None`. It also seems like an odd interface to distinguish between an empty list and `None`